Emby.Server.Implementations/ScheduledTasks/Tasks/MediaSegmentExtractionTask.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Data.Enums;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Dto;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.Tasks;
+
+namespace Emby.Server.Implementations.ScheduledTasks.Tasks;
+
+/// <summary>
+/// Task to obtain media segments.
+/// </summary>
+public class MediaSegmentExtractionTask : IScheduledTask
+{
+    /// <summary>
+    /// The library manager.
+    /// </summary>
+    private readonly ILibraryManager _libraryManager;
+    private readonly ILocalizationManager _localization;
+    private readonly IMediaSegmentManager _mediaSegmentManager;
+    private static readonly BaseItemKind[] _itemTypes = { BaseItemKind.Episode, BaseItemKind.Movie };
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MediaSegmentExtractionTask" /> class.
+    /// </summary>
+    /// <param name="libraryManager">The library manager.</param>
+    /// <param name="localization">The localization manager.</param>
+    /// <param name="mediaSegmentManager">The segment manager.</param>
+    public MediaSegmentExtractionTask(ILibraryManager libraryManager, ILocalizationManager localization, IMediaSegmentManager mediaSegmentManager)
+    {
+        _libraryManager = libraryManager;
+        _localization = localization;
+        _mediaSegmentManager = mediaSegmentManager;
+    }
+
+    /// <inheritdoc/>
+    public string Name => _localization.GetLocalizedString("TaskExtractMediaSegments");
+
+    /// <inheritdoc/>
+    public string Description => _localization.GetLocalizedString("TaskExtractMediaSegmentsDescription");
+
+    /// <inheritdoc/>
+    public string Category => _localization.GetLocalizedString("TasksLibraryCategory");
+
+    /// <inheritdoc/>
+    public string Key => "TaskExtractMediaSegments";
+
+    /// <inheritdoc/>
+    public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Should also add a cancellation check within the loop so the task can actually be cancelled between items